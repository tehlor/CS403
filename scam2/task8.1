(define t0 (newBST 5))
(displayBST t0)
