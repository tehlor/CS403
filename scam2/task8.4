(define t1 (newBST 47))
(define t2 (insertBST t1 22))
(define t3 (insertBST t2 71))
(define t4 (insertBST t3 93))
(define t5 (insertBST t4 92))
(define t6 (insertBST t5 37))
(define t7 (insertBST t6 43))
(define t8 (insertBST t7 48))
(define t9 (insertBST t8 35))
(define t10 (insertBST t9 68))
(define t11 (insertBST t10 39))
(define t12 (insertBST t11 61))
(define t13 (insertBST t12 90))
(define t14 (insertBST t13 57))
(define t15 (insertBST t14 87))
(define t16 (insertBST t15 62))
(define t17 (insertBST t16 19))
(define t18 (insertBST t17 33))
(define t19 (insertBST t18 97))
(define t20 (insertBST t19 86))
(displayBST t20)
