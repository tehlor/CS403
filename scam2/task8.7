(define t1 (newBST 25))
(define t2 (insertBST t1 32))
(define t3 (insertBST t2 47))
(define t4 (insertBST t3 86))
(define t5 (insertBST t4 22))
(define t6 (insertBST t5 69))
(define t7 (insertBST t6 54))
(define t8 (insertBST t7 42))
(define t9 (insertBST t8 13))
(define t10 (insertBST t9 43))
(define t11 (insertBST t10 46))
(define t12 (insertBST t11 76))
(define t13 (insertBST t12 29))
(define t14 (insertBST t13 96))
(define t15 (insertBST t14 34))
(define t16 (insertBST t15 17))
(define t17 (insertBST t16 49))
(define t18 (insertBST t17 12))
(define t19 (insertBST t18 19))
(define t20 (insertBST t19 36))
(define t21 (insertBST t20 50))
(define t22 (insertBST t21 66))
(define t23 (insertBST t22 67))
(define t24 (insertBST t23 73))
(define t25 (insertBST t24 51))
(define t26 (insertBST t25 62))
(define t27 (insertBST t26 15))
(define t28 (insertBST t27 89))
(define t29 (insertBST t28 63))
(define t30 (insertBST t29 74))
(define t31 (insertBST t30 75))
(define t32 (insertBST t31 80))
(define t33 (insertBST t32 55))
(define t34 (insertBST t33 56))
(define t35 (insertBST t34 88))
(define t36 (insertBST t35 33))
(define t37 (insertBST t36 37))
(define t38 (insertBST t37 83))
(define t39 (insertBST t38 68))
(define t40 (insertBST t39 99))
(define t41 (insertBST t40 57))
(define t42 (insertBST t41 84))
(define t43 (insertBST t42 11))
(define t44 (insertBST t43 14))
(define t45 (insertBST t44 78))
(define t46 (insertBST t45 21))
(define t47 (insertBST t46 82))
(define t48 (insertBST t47 58))
(define t49 (insertBST t48 70))
(define t50 (insertBST t49 26))
(define t51 (insertBST t50 40))
(define t52 (insertBST t51 27))
(define t53 (insertBST t52 87))
(define t54 (insertBST t53 72))
(define t55 (insertBST t54 28))
(define t56 (insertBST t55 92))
(define t57 (insertBST t56 31))
(define t58 (insertBST t57 35))
(define t59 (insertBST t58 61))
(define t60 (insertBST t59 93))
(displayBST t60)
