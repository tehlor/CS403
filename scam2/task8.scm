(define (main)
    (setPort (open (getElement ScamArgs 1) 'read))
    (readArgs)
)

(define (readArgs)
    (define env this)
    (define (iter expr)
        (if (not (eof?))
            (begin (eval expr env) 
                (iter (readExpr))
            )
        )
    )
    (iter (readExpr))
)

(define (node value left right)
    (define (display) (print value))
    this
)

(define (newBST value)
    (node value nil nil)
)

(define (displayBST root)
    (define (iter root indent)
        (if (valid? root)
            (begin
                (iter (root'right) (string+ indent "    "))
                (print indent)
                ((root'display))
                (println)
                (iter (root'left) (string+ indent "    "))
                )
            )
        )
    (iter root "")
)

(define (insertBST tree num)
    (if (< num (tree'value))
        (if (null? (tree'left))
            (node (tree'value) (node num nil nil) (tree'right))
            (node (tree'value) (insertBST (tree'left) num) (tree'right))
        )
        (if (> num (tree'value))
            (if (null? (tree'right))
                (node (tree'value) (tree'left) (node num nil nil))
                (node (tree'value) (tree'left) (insertBST (tree'right) num))
            )
        )
    )
)

(define ())