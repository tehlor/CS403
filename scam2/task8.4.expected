            97
        93
            92
                90
                    87
                        86
    71
            68
                    62
                61
                    57
        48
47
            43
                39
        37
            35
                33
    22
        19
